src/engine/SCons/Script/Main.py

```diff
@@ -712,54 +712,89 @@ def _load_site_scons_dir(topdir, site_dir_name=None):
+                    # SCons.Script module, and the exec ... in call executes a
+                    # file (or string containing code) in the context of the
+                    # module's dictionary, so anything that code defines ends
+                    # up adding to that module.  This is really short, but all
+                    # the error checking makes it longer.
+                    try:
+                        m = sys.modules['SCons.Script']
+                    except Exception as e:
+                        fmt = 'cannot import site_init.py: missing SCons.Script module {}'
+                        raise SCons.Errors.InternalError(fmt.format(repr(e)))
+                    try:
+                        sfx = description[0]
+                        modname = os.path.basename(pathname)[:-len(sfx)]
+                        site_m = {"__file__": pathname, "__name__": modname, "__doc__": None}
+                        re_special = re.compile("__[^_]+__")
+                        for k in list(m.__dict__.keys()):
+                            if not re_special.match(k):
+                                site_m[k] = m.__dict__[k]
+
+                        # This is the magic.
+                        exec(compile(fp.read(), fp.name, 'exec'), site_m)
+                    except KeyboardInterrupt:
+                        raise
+                    except Exception as e:
+                        fmt = '*** Error loading site_init file %s:\n'
+                        sys.stderr.write(fmt % repr(site_init_file))
+                        raise
+                    else:
+                        for k in site_m:
+                            if not re_special.match(k):
+                                m.__dict__[k] = site_m[k]
+                except KeyboardInterrupt:
+                    raise
+                except ImportError as e:
+                    fmt = '*** cannot import site init file %s:\n'
+                    sys.stderr.write(fmt % repr(site_init_file))
+                    raise
+            finally:
+                if fp:
+                    fp.close()
+    else:  # Python >= 3.5
+        if os.path.exists(site_init_file):
+            import importlib.util, importlib.machinery
             try:
-                fp, pathname, description = imp.find_module(site_init_modname,
-                                                            [site_dir])
-                # Load the file into SCons.Script namespace.  This is
-                # opaque and clever; m is the module object for the
-                # SCons.Script module, and the exec ... in call executes a
-                # file (or string containing code) in the context of the
-                # module's dictionary, so anything that code defines ends
-                # up adding to that module.  This is really short, but all
-                # the error checking makes it longer.
+                loader_details = (
```

Comment:
Please add a brief comment as to what this does

Reply:
Well, the same what the previous code ... opens module found in the particular location. See https://stackoverflow.com/a/37124336/164233 

OK, I will mention it.

Reply:
Also, this doesn't work.
```
[  106s] + python3 bootstrap.py build/scons
[  106s] /usr/bin/python3 /home/abuild/rpmbuild/BUILD/scons-3.0.1/bootstrap/src/script/scons.py build/scons
[  107s] *** Error loading site_init file './site_scons/site_init.py':
[  107s] AttributeError: 'NoneType' object has no attribute 'loader':
[  107s]   File "/home/abuild/rpmbuild/BUILD/scons-3.0.1/bootstrap/src/engine/SCons/Script/Main.py", line 1401:
[  107s]     _exec_main(parser, values)
[  107s]   File "/home/abuild/rpmbuild/BUILD/scons-3.0.1/bootstrap/src/engine/SCons/Script/Main.py", line 1364:
[  107s]     _main(parser)
[  107s]   File "/home/abuild/rpmbuild/BUILD/scons-3.0.1/bootstrap/src/engine/SCons/Script/Main.py", line 991:
[  107s]     _load_all_site_scons_dirs(d.get_internal_path())
[  107s]   File "/home/abuild/rpmbuild/BUILD/scons-3.0.1/bootstrap/src/engine/SCons/Script/Main.py", line 844:
[  107s]     _load_site_scons_dir(d)
[  107s]   File "/home/abuild/rpmbuild/BUILD/scons-3.0.1/bootstrap/src/engine/SCons/Script/Main.py", line 782:
[  107s]     mod = importlib.util.module_from_spec(spec)
[  107s]   File "<frozen importlib._bootstrap>", line 580:
```